QKSMS/src/main/java/com/moez/QKSMS/ui/view/ComposeView.java

```diff
@@ -753,7 +755,40 @@ public void saveDraft() {
 
                     // Only show the draft if we saved text, not if we just cleared some
                     if (!TextUtils.isEmpty(draft)) {
-                        Toast.makeText(mContext, R.string.toast_draft, Toast.LENGTH_SHORT).show();
+                        if (mRecipientProvider != null) {
+                            String[] addresses = mRecipientProvider.getRecipientAddresses();
+
+                            if (addresses != null && addresses.length > 0) {
+                                // save the message for each of the addresses
+                                for (int i = 0; i < addresses.length; i++) {
+                                    Calendar cal = Calendar.getInstance();
+                                    ContentValues values = new ContentValues();
+                                    values.put("address", addresses[i]);
+                                    values.put("date", cal.getTimeInMillis() + "");
+                                    values.put("read", 1);
+                                    values.put("type", 4);
+
+                                    // attempt to create correct thread id
+                                    long threadId = Utils.getOrCreateThreadId(mContext, addresses[i]);
+
+                                    Log.v(TAG, "saving message with thread id: " + threadId);
+
+                                    values.put("thread_id", threadId);
+                                    Uri messageUri = mContext.getContentResolver().insert(Uri.parse("content://sms/draft"), values);
+
+                                    Log.v(TAG, "inserted to uri: " + messageUri);
+
+                                    ConversationLegacy mConversationLegacy = new ConversationLegacy(mContext, threadId);
+                                    mConversationLegacy.saveDraft(draft);
+                                }
+                            } else {
+                                Toast.makeText(
```

Comment:
Can we make this just one line?

The `mRes.getString(R.string.error_no_recipients)` can also be replaced with `R.string.error_no_recipients`